core/common/src/main/java/alluxio/metrics/MetricsSystem.java

```diff
@@ -151,12 +152,11 @@ public static InstanceType fromString(String text) {
     METRIC_REGISTRY.registerAll(new ClassLoadingGaugeSet());
     METRIC_REGISTRY.registerAll(new CachedThreadStatesGaugeSet(5, TimeUnit.SECONDS));
     METRIC_REGISTRY.registerAll(new OperationSystemGaugeSet());
-  }
 
-  public static final BufferPoolMXBean DIRECT_BUFFER_POOL;
-
-  static {
     DIRECT_BUFFER_POOL = getDirectBufferPool();
+    MetricsSystem.registerGaugeIfAbsent(
+        MetricsSystem.getMetricName(MetricKey.POOL_DIRECT_MEM_USED.getName()),
+        MetricsSystem::getDirectMemUsed);
```

Comment:
Will it be heavy to get the direct mem used? since gauge can fetch direct mem used every second. If it needs calculation, could be heavy. If just get a existing number then it's fine

also, getDirectMemUsed method probably better to not put in metrics system class directly since this metrics system is complicated enough, and that metric can be registered in other classes.
so that future system-related metrics can also be added to that class.

Reply:
It is an atomic:
![image](https://user-images.githubusercontent.com/39251482/192940072-ba47c11d-acb4-4bf7-b4e2-c9bdcd084057.png)
![image](https://user-images.githubusercontent.com/39251482/192940100-7c3b45ba-2ef9-41a0-b10c-3f532f51a6cf.png)

Reply:
If it is not put into the metrics system, a dedicate class should be created for that. 
Will move that if a better class is found in the future.